src/wallet/db.h

```diff
@@ -156,6 +182,11 @@ class WalletDatabase
     virtual std::unique_ptr<DatabaseBatch> MakeBatch(bool flush_on_close = true) = 0;
 };
 
+class DummyCursor : public DatabaseCursor
+{
+    bool Next(CDataStream& key, CDataStream& value, bool& complete) override { return false; }
```

Comment:
Check this https://github.com/bitcoin/bitcoin/pull/24914#discussion_r1027288086.
As we usually run this code inside a loop until it fails or return `complete=true`, if it never fails, then it can become a bug in the future.